src/insns/modesw_common.adoc

```diff
@@ -5,10 +5,13 @@ Toggle the hart's current CHERI execution mode in <<pcc>>.
 * If the current mode in <<pcc>> is {cheri_int_mode_name} ({INT_MODE_VALUE}), then the <<m_bit>> in <<pcc>> is set to {cheri_cap_mode_name} ({CAP_MODE_VALUE}).
 * If the current mode is {cheri_cap_mode_name} ({CAP_MODE_VALUE}), then the <<m_bit>> in <<pcc>> is set to {cheri_int_mode_name} ({INT_MODE_VALUE}).
 
-NOTE: The effective CHERI exection mode is give by the value of some CSRs and
+NOTE: The effective CHERI execution mode is give by the value of some CSRs and
 the <<pcc>>'s <<m_bit>>, so executing <<MODESW>> does not necessarily change
 the machine's current mode. The current, effective CHERI execution mode can be
 observed as described in xref:m_bit_observe[xrefstyle=short].
 
-NOTE: Implementations may optionally support executing <<C_MODESW>> from the
-program buffer while in debug mode.
+NOTE: Implementations may optionally support executing <<MODESW>> from the
+program buffer while in debug mode. If supported them the <<m_bit>> in
+<<dinfc>> is toggled and used to control which mode to enter next time debug
+mode is entered. The CHERI execution mode is only controlled by the <<m_bit>>
+of <<dinfc>> in debug mode.
```

Comment:
I am concerned that MODESW is allowed when the PCC is unspecified because:
1. We already have a way easily change the mode by read-modify-write dinfc using csrr, scmode, csrw
2. Changes the instruction's behaviour just for this privilege mode

Also, note that the debug spec also allows having the PCC defined properly, so that the program buffer is part of the memory address space allowing jumps, branches, auipc, etc to be executed from there. What determines the CHERI mode in this case? If its the PCC then do we need to update dinfc whenever PCC is written with (e.g.) JALR/MODESW turning the M bit across these two registers into some WARL-like field?

It would be best to disallow MODESW while in debug mode altogether and simply have the mode there be determined by dinfc in all cases.